api/v1beta1/temporalworkerprocess_types.go

```diff
@@ -0,0 +1,100 @@
+// Licensed to Alexandre VILAIN under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Alexandre VILAIN licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v1beta1
+
+import (
+	"fmt"
+
+	"github.com/alexandrevilain/temporal-operator/pkg/version"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TemporalWorkerProcessSpec defines the desired state of TemporalWorkerProcess
+type TemporalWorkerProcessSpec struct {
+	// Reference to the temporal cluster the namespace will be created.
+	ClusterRef *TemporalClusterReference `json:"clusterRef"`
+	// Version defines the worker process version.
+	// +optional
+	Version *version.Version `json:"version,omitempty"`
```

Comment:
Hum, I created the `Version` struct to manage temporal versionning.
But if the users want to use, for instance, an docker image tag named `dev-feature-1`, it will break.
IMO it would be safer to ask for a string here. 
```suggestion
	Version string `json:"version,omitempty"`
```

Reply:
@alexandrevilain The problem is if I go with string it breaks GetLabels so I would need another GetLabels func since it grabs version but sure I will add another GetLabels.